src/view/components/DataTable/index.tsx

```diff
@@ -1,11 +1,14 @@
 import * as React from 'react';
 import { List, ListItem, Text, Grid, Col, Card } from 'native-base';
 import styles from './style';
+import { Route } from '../../navigation/navigator';
 
 
 interface OwnPropsType {
   headers: string[];
   values: any[];
+  route?: Route;
+  onPress?(route: Route): void;

```

Comment:
If one of these being required makes it necessary the other is required too, you can do one of two things:
1) Create an interface that holds a required route and onPress that is itself an optional property on onPress
2) Create an interface (`RoutedPropsType` for this example) that extends the interface with headers/values (we will call this `BasePropsType` for my example) and then `type OwnPropsType = RoutedPropsType | BasePropsType`

Reply:
ooh that's clever! done.